topics/basics/testing_plugins/integration_tests/integration_tests_intro.md

```diff
@@ -0,0 +1,262 @@
+* Full product testing: Integration tests run against the complete product rather than isolated components. This helps identify issues that unit tests might miss, such as module interaction problems, classpath conflicts, and plugin declaration issues.
+* User story validation: Integration tests typically mirror real user scenarios, ensuring your plugin works reliably from start to finish.
+
+Despite common perceptions, integration tests aren't as difficult to create as you might think! Let's explore how to set them up.
+
+## Adding dependencies
+Integration testing framework consists of two main components:
+* **Starter**: Handles IDE configuration, test project setup, IDE startup, and output collection.
+* **Driver**: Provides additional functionality we'll explore in future blog posts.
+
+The Starter framework exclusively supports JUnit 5, as it leverages JUnit 5's extensions and specialized listeners that aren't available in JUnit 4.
+
+Add the following dependencies to your build.gradle.kts:
+```kotlin
+dependencies {
+    intellijPlatform {
+        //...
+        testFramework(TestFrameworkType.Starter)
+    }
+
+    testImplementation("org.junit.jupiter:junit-jupiter:5.10.2")
+    testImplementation("org.kodein.di:kodein-di-jvm:7.20.2")
+}
+```
+or if you don't use IntelliJ Gradle Platform Plugin:
+```
+// Starter Framework
+testImplementation("com.jetbrains.intellij.tools:ide-starter-squashed:LATEST-EAP-SNAPSHOT")
+testImplementation("com.jetbrains.intellij.tools:ide-starter-junit5:LATEST-EAP-SNAPSHOT")
+
+// Driver Framework
+testImplementation("com.jetbrains.intellij.tools:ide-starter-driver:LATEST-EAP-SNAPSHOT")
+testImplementation("com.jetbrains.intellij.driver:driver-client:LATEST-EAP-SNAPSHOT")
+testImplementation("com.jetbrains.intellij.driver:driver-sdk:LATEST-EAP-SNAPSHOT")
+testImplementation("com.jetbrains.intellij.driver:driver-model:LATEST-EAP-SNAPSHOT")
+
+// JUnit 5
+testImplementation("org.junit.jupiter:junit-jupiter:5.10.2")
+// Dependency Injection for Starter configuration
+testImplementation("org.kodein.di:kodein-di-jvm:7.20.2")
+```
+
+Note: While `LATEST-EAP-SNAPSHOT` is available, you might prefer using specific release versions to ensure test stability. The framework follows the IntelliJ IDEA versioning scheme, for example:
+`testImplementation("com.jetbrains.intellij.tools:ide-starter-squashed:243.23654.153")`.
+
+Note:
+* Use the same version across all framework libraries to avoid compatibility issues.
+* As the Driver and UI components continue to evolve, we aim to keep the Starter API stable, with occasional breaking changes.
+
+## Integrating Starter with IntelliJ Platform Gradle Plugin
+
+To test your plugin, the Starter framework needs to know where to find your plugin distribution for installation in the IDE. This requires configuring your Gradle test task.
+
+Add the following configuration to your `build.gradle.kts`:
+```
+tasks.test {
```

Comment:
is it intended to use `test` target here instead of introducing dedicated `integrationTest` or similar?

Reply:
All our examples as well as blog posts, lectures and talks are using `test` task. I would like to keep it that way to avoid discrepancies between different sources.

Reply:
AFAIU the default setup for Gradle is this https://docs.gradle.org/current/userguide/java_testing.html#sec:configuring_java_integration_tests